components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/util/MetadataConverter.java

```diff
@@ -20,22 +20,26 @@
 import org.wso2.carbon.identity.application.common.model.Property;
 import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
 import org.wso2.carbon.idp.mgt.IdentityProviderSAMLException;
+
 /**
  * This interface provides functionality to convert a string to federatedAuthenticatorConfig and vise versa
  */
 public interface MetadataConverter {
 
     boolean canHandle(Property property);
 
-    FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(Property properties [], StringBuilder builder)
+    FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(Property[] properties, StringBuilder builder)
             throws IdentityProviderManagementException, javax.xml.stream.XMLStreamException;//TODO
 
     String getMetadataString(FederatedAuthenticatorConfig federatedAuthenticatorConfig) throws IdentityProviderSAMLException;
 
     boolean canHandle(FederatedAuthenticatorConfig federatedAuthenticatorConfig);
 
-    void saveMetadataString(int tenantId, String idpName, String metadata) throws IdentityProviderManagementException;
+    void saveMetadataString(int tenantId, String idpName, String fedAuthName, String metadata) throws IdentityProviderManagementException;
 
     void deleteMetadataString(int tenantId, String idPName)throws IdentityProviderManagementException;
 
-}
\ No newline at end of file
+    default boolean canDelete(int tenantId, String idPName) throws IdentityProviderManagementException{
```

Comment:
Shall we add a newline after the method signature? Also, can we add a method comment?